fixtures/ios_app_with_templates/Tuist/Templates/custom/Template.swift

```diff
@@ -0,0 +1,23 @@
+import ProjectDescription
+
+let nameAttribute: Template.Attribute = .required("name")
+let platformAttribute: Template.Attribute = .optional("platform", default: "ios")
+
+let testContents = """
+// this is test \(nameAttribute) content
+"""
+
+let template = Template(
+    description: "Custom template",
+    attributes: [
+        nameAttribute,
+        platformAttribute
+    ],
+    files: [
```

Comment:
I looked at this file before the code and the documentation to see if I'd be able to understand it, and these are the things that popped in my mind:

- What do `directories` mean? 
- Why is the file `static` when it's dynamically generated when scaffolding?
- What does `generateFilePath` mean?
- Why `custom_dir` appears both in the `files` and also in the `directories` attribute?

Reply:
`directories` is a list of directories that should be generated - this has two implications: creating directories is an explicit operation and is not hidden in a file structure of the files. That means that if user wants to quickly see what directories will be generated, this is the one place they can look at. As a side benefit, users can create empty directories (though it is arguable how useful it is ... ) But we can also derive the generated directories from the files, which would make the setup a little bit simpler.
To summarize, do you feel like that property should be renamed or dropped altogether?